### PR TITLE
Added a new section for mount options

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://trello.com/c/zI33xUyz/419-as-a-openshift-user-i-should-be-able-to-use-mount-options

### CODE REVIEW

File: architecture/additional_concepts/storage.adoc [hunk 1]
```diff
@@ -405,6 +405,43 @@ cluster.
...
+----
+<1> Specified mount options are then used while mounting the persistent volume to the disk.
+====
+
+[NOTE]
+====
+Fiber Channel and HostPath persistent volumes do not support mount options.
+====
+
 [[persistent-volume-claims]]
```
Comment (gnufied) [inline architecture/additional_concepts/storage.adoc]: Should this be rather more exhaustive list of volume types that supports `mount-options` ? Something like - https://kubernetes.io/docs/concepts/storage/persistent-volumes/#mount-options

Comment (gaurav-nelson) [inline architecture/additional_concepts/storage.adoc]: Thanks @gnufied 
I have updated the PR to include the list of supported PVs.

**Comment by gaurav-nelson** [comment]
[Document: As a Openshift user I should be able to use mount options](https://trello.com/c/QHfh8YJA/466-document-as-a-openshift-user-i-should-be-able-to-use-mount-options)

**Comment by gnufied** [comment]
lgtm!

**Comment by gaurav-nelson** [comment]
Updated to include:

- Tech Preview warning
- only available for manually provisioned PVs

**Comment by gaurav-nelson** [comment]
@bfallonf peer review please

File: architecture/additional_concepts/storage.adoc [hunk 2]
```diff
@@ -405,6 +405,70 @@ cluster.
...
+      storage: 1Gi
+    accessModes:
+    - ReadWriteOnce
+    nfs:
+      path: /tmp
+      server: 172.17.0.2
+    persistentVolumeReclaimPolicy: Recycle
+    claimRef:
+      name: claim1
+      namespace: default
```
Comment (bfallonf) [inline architecture/additional_concepts/storage.adoc]: @gaurav-nelson I'm not sure, but will the above being one space out from the edge cause problems?

**Comment by bfallonf** [comment]
@gaurav-nelson One comment from me. Looks good otherwise :100:

File: architecture/additional_concepts/storage.adoc (hunk 2 above)
Comment (gaurav-nelson) [inline architecture/additional_concepts/storage.adoc]: Thanks @bfallonf It rendered fine but I have fixed the extra spaces to be consistent.

**Comment by gaurav-nelson** [comment]
[rev_history]
|xref:../architecture/additional_concepts/storage.adoc#architecture-additional-concepts-storage[Persistent Storage]
|Added xref:../architecture/additional_concepts/storage.adoc#pv-mount-options[Mount Options] section.
%

**Pull request merged**